drivers/sensor/st/lsm6dsv16x/lsm6dsv16x_rtio_stream.c

```diff
@@ -0,0 +1,363 @@
+/* ST Microelectronics LSM6DSV16X 6-axis IMU sensor driver
+ *
+ * Copyright (c) 2023 Google LLC
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_lsm6dsv16x
+
+#include <zephyr/dt-bindings/sensor/lsm6dsv16x.h>
+#include <zephyr/drivers/sensor.h>
+#include "lsm6dsv16x.h"
+#include "lsm6dsv16x_decoder.h"
+#include <zephyr/rtio/work.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(LSM6DSV16X_RTIO);
+
+#define FIFO_TH 1
+#define FIFO_FULL 2
+
+static void lsm6dsv16x_config_fifo(const struct device *dev, uint8_t fifo_irq)
+{
+	struct lsm6dsv16x_data *lsm6dsv16x = dev->data;
+	const struct lsm6dsv16x_config *config = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&config->ctx;
+	uint8_t fifo_wtm = 0;
+	lsm6dsv16x_pin_int_route_t pin_int = { 0 };
+	lsm6dsv16x_fifo_xl_batch_t xl_batch = LSM6DSV16X_DT_XL_NOT_BATCHED;
+	lsm6dsv16x_fifo_gy_batch_t gy_batch = LSM6DSV16X_DT_GY_NOT_BATCHED;
+	lsm6dsv16x_fifo_temp_batch_t temp_batch = LSM6DSV16X_DT_TEMP_NOT_BATCHED;
+	lsm6dsv16x_fifo_mode_t fifo_mode = LSM6DSV16X_BYPASS_MODE;
+
+	/* disable FIFO as first thing */
+	lsm6dsv16x_fifo_mode_set(ctx, LSM6DSV16X_BYPASS_MODE);
+
+	pin_int.fifo_th = PROPERTY_DISABLE;
```

Comment:
should you also be setting `pin_int.fifo_full = PROPERTY_DISABLE`

Reply:
... oooops, correct!

Reply:
I'll submit a new commit set on Monday. I'm leaving for the week

Reply:
Done. We should be all set I guess